lib/src/provider.dart

```diff
@@ -201,33 +214,17 @@ class StatefulProvider<T> extends StatefulWidget
     implements SingleChildCloneableWidget {
   /// Allows to specify parameters to [StatefulProvider]
   StatefulProvider({
-    Key key,
     @required this.valueBuilder,
+    Key key,
     this.child,
     this.dispose,
     this.updateShouldNotify,
   })  : assert(valueBuilder != null),
         super(key: key);
 
-  /// A function that creates the provided value.
-  ///
-  /// [valueBuilder] must not be null and is called only once for the life-cycle of [StatefulProvider].
-  ///
-  /// It is not possible to obtain an [InheritedWidget] from [valueBuilder].
+  final Widget child;
   final ValueBuilder<T> valueBuilder;
-
-  /// [dispose] is a callback called when [StatefulProvider] is
-  /// removed for the widget tree, and pass the current value as parameter.
-  /// It is useful when the provided object needs to have a custom dipose behavior,
-  /// such as closing streams.
   final Disposer<T> dispose;
-
-  /// The widget below this widget in the tree.
-  ///
-  /// {@macro flutter.widgets.child}
```

Comment:
Out of curiosity, why remove these pieces of documentation?

Reply:
Mostly because the current documentation is redundant - the class-level comment has good coverage. I can keep them if you prefer though. I don't have a strong opinion here.

Reply:
I actually have a question here:

These members are public - is this necessary? I see that some of the test code accesses them. In my opinion these fields can as well be marked package-private (with the underscore prefix). WDYT?

Reply:
Documenting every single public field is necessary, otherwise pub decreases the health score of the package.

> these fields can as well be marked package-private

I'm not sure. 
I've never seen a widget hiding its fields.

Reply:
Added comments for those fields.

I've seen them :-)

Nevertheless, this is not a big issue. And I do not want to block this change.